File: backend/app/services/auth_service.py
from fastapi import HTTPException, status
from sqlalchemy.orm import Session

from app.schemas.user import UserCreate, UserLogin
from app.repositories.user_repository import UserRepository
from app.core.security import hash_password, verify_password, create_access_token

class AuthService:

    @staticmethod
    def register(db: Session, payload: UserCreate):
        existing_user = UserRepository.get_by_email(db, payload.email)

        if existing_user:
            raise HTTPException(
                status_code=status.HTTP_409_CONFLICT,
                detail="Bu email zaten kayıtlı",
            )
        
        password_hash = hash_password(payload.password)

        user = UserRepository.create(
            db=db,
            full_name=payload.full_name,
            email=payload.email,
            password_hash=password_hash,
        )

        return user

    @staticmethod
    def login(db: Session, payload: UserLogin):
        user = UserRepository.get_by_email(db, payload.email)

        if not user:
            raise HTTPException(
                status_code=status.HTTP_401_UNAUTHORIZED,
                detail="Email veya şifre hatalı",
            )
        
        if not verify_password(payload.password, user.password_hash):
            raise HTTPException(
                status_code=status.HTTP_401_UNAUTHORIZED,
                detail="Email veya şifre hatalı.",
            )
        
        access_token = create_access_token(subject=str(user.id))

        return {
            "access_token": access_token,
            "token_type": "bearer",
            "current_user": user,
        }

File: frontend/assets/js/services/authService.js
import { apiClient } from "../core/apiClient.js";
import { storage } from "../core/storage.js";

export const authService = {
  async login(email, password) {
    const data = await apiClient.post(
      "/api/v1/auth/login",
      {
        email,
        password,
      },
      {
        auth: false,
      }
    );

    storage.setToken(data.access_token);

    return data;
  },

  async register(full_name, email, password) {
    return await apiClient.post(
      "/api/v1/auth/register",
      {
        full_name,
        email,
        password,
      },
      {
        auth: false,
      }
    );
  },

  async getCurrentUser() {
    return await apiClient.get("/api/v1/auth/me");
  },

  logout() {
    storage.removeToken();
  },
};